lib/matplotlib/collections.py

```diff
@@ -545,9 +545,12 @@ def set_offsets(self, offsets):
         ----------
         offsets : (N, 2) or (2,) array-like
         """
-        offsets = np.asanyarray(offsets, float)
+        offsets = np.asanyarray(offsets)
         if offsets.shape == (2,):  # Broadcast (2,) -> (1, 2) but nothing else.
             offsets = offsets[None, :]
+        offsets[:, 0] = self.convert_xunits(offsets[:, 0])
+        offsets[:, 1] = self.convert_yunits(offsets[:, 1])
+        offsets = np.asarray(offsets, 'float')
```

Comment:
If we're going to re-cast, is there any reason to hang on to the original array? Can we do `np.column_stack((convertx(...), converty(...)))`?

Reply:
sure, done.  Not sure it is any more elegant, but probably better to convert the x and y arrays separately to float anyway (they can be object arrays, even though convert_xunits has been called, which actually strikes me as a problem in convert_x/yunits.